ruby/src/main/java/com/google/protobuf/jruby/RubyMessage.java

```diff
@@ -345,13 +345,18 @@ public IRubyObject toHash(ThreadContext context) {
         for (Descriptors.FieldDescriptor fdef : this.descriptor.getFields()) {
             IRubyObject value = getField(context, fdef);
             if (!value.isNil()) {
-                if (value.respondsTo("to_h")) {
+                // Order here matters!  Enumerable fields, such as RepeatedField,
+                // respond to both #to_h and #to_a.  in that case, we want the #to_a
+                // behavior
```

Comment:
this is an issue under jruby-9000, but not MRI ruby or earlier versions of jruby